plugins/connection/nsenter.py

```diff
@@ -0,0 +1,259 @@
+# (c) 2021 Jeff Goldschrafe <jeff@holyhandgrenade.org>
+# Based on Ansible local connection plugin by:
+# (c) 2012 Michael DeHaan <michael.dehaan@gmail.com>
+# (c) 2015, 2017 Toshio Kuratomi <tkuratomi@ansible.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    name: nsenter
+    short_description: execute on host running controller container
+    version_added: 1.9.0
+    description:
+        - This connection plugin allows Ansible, running in a privileged container, to execute tasks on the container
+          host instead of in the container itself.
+        - This is useful for running Ansible in a pull model, while still keeping the Ansible control node
+          containerized.
+        - It relies on having privileged access to run nsenter in the host's PID namespace, allowing it to enter the
+          namespaces of the provided PID (default PID 1, or init/systemd).
+    author: Jeff Goldschrafe (@jgoldschrafe)
+    options:
+        host_volume_mount:
+            description:
+                - Host volume mount to read and write host files through.
+            type: string
+            default: /host
+            vars:
+                - name: ansible_host_volume_mount
+            env:
+                - name: ANSIBLE_HOST_VOLUME_MOUNT
```

Comment:
I might recommend adding nsenter into these names.
```suggestion
            vars:
                - name: ansible_nsenter_host_volume_mount
            env:
                - name: ANSIBLE_NSENTER_HOST_VOLUME_MOUNT
```

Reply:
One better: I funneled the file I/O through `nsenter` as well, and made the volume mounts unnecessary. Nice flag, thank you!